src/script/auth/page/SetEntropy.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CheckRoundIcon, ContainerXS, H1, Muted, Text} from '@wireapp/react-ui-kit';
+import React, {MouseEvent, PointerEvent, useCallback, useEffect, useState} from 'react';
+
+import {usePausableInterval} from '../../hooks/usePausableInterval';
+import {usePausableTimeout} from '../../hooks/usePausableTimeout';
+import {setEntropyStrings} from '../../strings';
+import {useIntl} from 'react-intl';
+import Canvas from '../component/Canvas';
+
+import {ProgressBar} from '../component/ProgressBar';
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  entropy: [number, number][];
+  error: any;
+  onSetEntropy: () => void;
+  setEntropy: React.Dispatch<React.SetStateAction<[number, number][]>>;
+  setError: React.Dispatch<any>;
+}
+
+const SetEntropy = ({setEntropy, entropy, error, setError, onSetEntropy}: Props) => {
+  const [frameCount, setFrameCount] = useState(0);
+  const [percent, setPercent] = useState(0);
+  const {formatMessage: _} = useIntl();
+
+  const onMouseMove = (event: MouseEvent<HTMLCanvasElement> | PointerEvent<HTMLCanvasElement>) => {
+    setError(null);
+    setFrameCount(frameCount + 1);
+
+    const newEntropy: [number, number] = [
+      event.pageX - event.currentTarget?.getBoundingClientRect()?.x,
+      event.pageY - event.currentTarget?.getBoundingClientRect()?.y,
+    ];
+    setEntropy(entropy => [...entropy, newEntropy]);
+  };
+
+  const draw = useCallback(
+    (ctx: CanvasRenderingContext2D) => {
+      if (entropy.length > 1 && frameCount > 1) {
+        ctx.beginPath();
+        if (!entropy[frameCount - 2]) {
+          ctx.moveTo(...entropy[frameCount - 1]);
+        } else {
+          ctx.moveTo(...entropy[frameCount - 2]);
+        }
+        ctx.lineJoin = 'round';
+        ctx.lineCap = 'round';
+        ctx.lineWidth = 2;
+        ctx.strokeStyle = 'blue';
+        if (!entropy[frameCount - 1]) {
+          ctx.lineTo(...entropy[frameCount - 2]);
+        } else {
+          ctx.lineTo(...entropy[frameCount - 1]);
+        }
+        ctx.stroke();
+      }
+    },
+    [entropy],
+  );
```

Comment:
I am not very comfortable with the relationship between `Canvas` and `SetEntropy`. 
Here `Canvas` is exposing some internal structure for the parent to draw in it, this kinds of defeat the purpose of isolating the canvas in its own component in the first place. 

I think we should find a way to completely hide the drawing part inside the `Canvas` component. 
We could specialize it a little and call it a `EntropyCanvas`. Its responsabilities will be:
- binding mouse in/out events and listen to them
- draw the lines according to those mouse movement inside of it
- warn the parent when a new position is drawn so that the parent can store it in the entropy array.

The API of the `EntropyCanvas` would then be:

```
<EntropyCanvas
            sizeX={}
            sizeY={}
            onEntropyData={position => storeEntropyData(position)}
          />
```

And we can get rid of `onMouseEnter`, `onMouseMove`, `onMouseLeave` and `draw`. They will be internals of the component. 

WDYT?

Reply:
i think this is a very good note, I will try to move to this direction tomorrow.